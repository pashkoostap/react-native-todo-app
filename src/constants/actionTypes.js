export const GET_TODOS_INIT = "GET_TODOS_INIT";
export const GET_TODOS_SUCCESS = "GET_TODOS_SUCCESS";
export const GET_TODOS_FAIL = "GET_TODOS_FAIL";
export const ADD_TODO = "ADD_TODO";
export const DELETE_TODO = "DELETE_TODO";
