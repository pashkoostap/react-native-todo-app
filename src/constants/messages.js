export const noTodosYetText = "There are no any todos yet";
