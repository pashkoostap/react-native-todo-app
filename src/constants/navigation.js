export const HOME = "Home";
export const TODO_MODAL = "TodoModal";
