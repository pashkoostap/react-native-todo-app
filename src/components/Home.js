import React from "react";
import { StyleSheet, View, Text, ScrollView } from "react-native";

import { noTodosYetText } from "../constants/messages";
import Todos from "./Todos";

const Home = ({ todos }) => (
  <View style={styles.container}>
    <ScrollView>
      {todos.length ? <Todos todos={todos} /> : <Text>{noTodosYetText}</Text>}
    </ScrollView>
  </View>
);

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: "center",
    justifyContent: "center"
  }
});

export default Home;
