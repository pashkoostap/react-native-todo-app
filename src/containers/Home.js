import React, { Component } from "react";
import { connect } from "react-redux";

import { getTodos } from "../store/actions/todos";
import { mapTodos } from "../models";

import Home from "../components/Home";

class HomeScreen extends Component {
  static navigationOptions = ({ navigation }) => ({
    headerTitle: "dasdas",
    header: false
  });

  componentDidMount() {
    this.props.getTodos();
  }

  render() {
    return <Home todos={this.props.todos} />;
  }
}

const mapStateToProps = ({ todosReducer }) => ({
  todos: mapTodos(todosReducer.todos)
});

const mapDispatchToProps = dispatch => ({
  getTodos: () => dispatch(getTodos())
});

export default connect(mapStateToProps, mapDispatchToProps)(HomeScreen);
