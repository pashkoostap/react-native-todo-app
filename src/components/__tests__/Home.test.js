import React from "react";
import { Text } from "react-native";
import renderer from "react-test-renderer";

import { noTodosYetText } from "../../constants/messages";
import Home from "../Home";

describe("Home", () => {
  it("should match the snapshot", () => {
    const todos = ["todo 1", "todo 2", "todo 3"];
    const rendered = renderer.create(<Home todos={todos} />).toJSON();

    expect(rendered).toMatchSnapshot();
  });

  it("shoud contain Text element if todos are the empty array", () => {
    const todos = [];
    const root = renderer.create(<Home todos={todos} />).root;
    const innerText = root.findByType(Text).props.children;

    expect(innerText).toBe(noTodosYetText);
  });
});
