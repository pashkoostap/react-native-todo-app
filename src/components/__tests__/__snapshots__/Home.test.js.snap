// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should match the snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <RCTScrollView>
    <View>
      <View>
        <View>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
          />
        </View>
        <View>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
          />
        </View>
        <View>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
          />
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
